### DISCUSSION TITLE
upload and download speed-test

### DISCUSSION THREAD

**Discussion by ToptenDev** [community/community#45725] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Nowadays I am trying to use fast.com to test upload and download speed test. But I found that I need to know the fast.com api token to have a test.
Is there any way to get it automatically with JavaScript?

**Comment by toprater** [comment]
I guess there was a similar question before.
https://github.com/branchard/fast-speedtest-api/issues/29

**Comment by Krieit** [comment, marked as answer]
1. Send a GET request to https://fast.com/
2. Parse the document using cheerio
3. Query the script tag and get its src
4. Fetch the script content
5. Parse the script with acorn
6. Find the token using acorn-walk
7. Send requests to https://api.fast.com/netflix/speedtest/v2?https=true&token=${token}&urlCount=5 to get the request object urls.
8. Send all requests at once.
9. Benchmark using stream.on('data').

You can get fast.com token using `fast.com` module.
https://www.npmjs.com/package/fast.com

**Reply by emekaorji** [reply]
This seems to only work for download speed, any pointers for testing upload speed?

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ToptenDev ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
